project changes

Tomich1 opened the pull request:
class object is created. fields are filled.

Tomich1 committed:
project changes

class object is created. fields are filled.

Ghostwish46 commented on .idea/.name:
```diff
@@ -0,0 +1 @@
+Novella3
```
По хорошему файлы из .idea папки - временные локальные и должны игнорироваться, но пока пропустим этот момент.

Ghostwish46 commented on app/src/main/java/com/example/novella3/ContentActivity.kt:
```diff
@@ -4,8 +4,15 @@ import androidx.appcompat.app.AppCompatActivity
+        setContentView(R.layout.activity_content)
+    }
+>>>>>>> Stashed changes
 }
```
Вот тут значок есть, что нет переноса на новую строку в конце файла, надо добавить. И в местах ниже тоже.

Ghostwish46 commented on app/src/main/java/com/example/novella3/MainActivity.kt:
```diff
@@ -12,13 +12,19 @@ class MainActivity : AppCompatActivity() {
-			
+
 		}
 	}
```
Убедись, пожалуйста, что тут нет излишних отступов (два и более переноса на новую строку, или переносы перед закрывающими скобками)

Ghostwish46 commented on app/src/main/java/com/example/novella3/MainActivity.kt:
```diff
@@ -12,13 +12,19 @@ class MainActivity : AppCompatActivity() {
 	override fun onCreate(savedInstanceState: Bundle?) {
 		super.onCreate(savedInstanceState)
 		setContentView(R.layout.activity_1)
```
Название класса активности должно перекликаться с файлом в xml разметке, чтобы проще ориентироваться. Судя по названию, тут должно быть activity_main

Ghostwish46 commented on app/src/main/java/com/example/novella3/MainActivity.kt:
```diff
@@ -12,13 +12,19 @@ class MainActivity : AppCompatActivity() {
 		setContentView(R.layout.activity_1)
+<<<<<<< Updated upstream
 		
 		val startGame = findViewById<Button>(R.id.bt1)
```
судя по названию переменной, кнопка должна быть не bt1, а хотя бы btStart / btStartGame

Ghostwish46 commented on app/src/main/java/com/example/novella3/MainActivity.kt:
```diff
@@ -12,13 +12,19 @@ class MainActivity : AppCompatActivity() {
+
+>>>>>>> Stashed changes
 		startGame.setOnClickListener {
 			val intent = Intent(this, MainActivity2::class.java)
```
MainActivity2 тоже не звучащее название, стоит переименовать

Ghostwish46 commented on app/src/main/java/com/example/novella3/NovellaAction.kt:
```diff
@@ -1,5 +1,7 @@
 package com.example.novella3
 
-class NovellaAction (val page : Int, val textButton : String){
+
+class NovellaAction (val page : Int, val text : String){
 
 }
```
Лишние пустоты?

Ghostwish46 replied:
Фигурные скобки тоже убрать стоит, раз там ничего нет

Ghostwish46 commented on app/src/main/java/com/example/novella3/NovellaContent.kt:
```diff
@@ -1,14 +1,11 @@
-	
-)
-
+    var imageResId: Int,
+    var text:String,
+    val firstAction: NovellaAction?,
+    val secondAction: NovellaAction?,
+    val thirdAction: NovellaAction?
 
+)
```
Пустоты

Ghostwish46 commented on app/src/main/java/com/example/novella3/Storage.kt:
```diff
@@ -0,0 +1,144 @@
+package com.example.novella3
+
+object Storage {
+
```
Есть сочетание клавиш Ctrl+Alt+L которое пытается отформатировать код. Советую в каждом файле его применить.

Ghostwish46 commented on app/src/main/java/com/example/novella3/Storage.kt:
```diff
@@ -0,0 +1,144 @@
+object Storage {
+
+    val pages:List<NovellaContent> = listOf(
+            //page 3
+        NovellaContent(
+            imageResId = R.drawable.third,
+            text = "Отлично, чем займёмся?",
+            firstAction = NovellaAction(page = 3, text = "пойдем на пробежку"),
+            secondAction = null,
+            thirdAction = null),
+        NovellaContent(
+            imageResId = R.drawable.third,
+            text = "Отлично, чем займёмся?",
+            secondAction = NovellaAction(page = 3, text = "пойдем в поход"),
+            firstAction = null,
+            thirdAction = null),
+        NovellaContent(
+            imageResId = R.drawable.third,
+            text = "Отлично, чем займёмся?",
+            thirdAction = NovellaAction(page = 3, text = "пойдем по полю"),
+            firstAction = null,
+            secondAction = null),
```
Предполагалось, что одна страница = один новый объект NovellaContent, его вполне достаточно, чтобы описать все тексты и кнопки.
Тут у тебя повторяются imageResId и text, а значит есть намёки, что это один объект. Просто в данном случае у тебя будут заполнены firstAction, secondAction  и thirdAction. Понимаешь?

Ghostwish46 commented on app/src/main/java/com/example/novella3/Storage.kt:
```diff
@@ -0,0 +1,144 @@
+            thirdAction = NovellaAction(page = 3, text = "пойдем по полю"),
+            firstAction = null,
+            secondAction = null),
+
```
Ниже тот же самый момент, что коммент сверху

Ghostwish46 commented on app/src/main/res/layout/activity_4.xml:
```diff
@@ -51,7 +51,7 @@
         android:layout_height="wrap_content"
         android:layout_marginBottom="26dp"
-        android:text="Да, и посмотрим фильм "
+        android:text="Да и посмотрим фильм "
```
Это пока не принципиально, но здесь и ниже правильнее оставить запятую)

Tomich1 committed:
Making changes after the review

disconnected files.idea
added a newline to the end of the file
removed extra padding
changed the names for the Start and Name Entry classes
changed the name of the button to start the game
Performed code formatting with hotkeys
Updated the description of the objects on each page in the Storage file
Added punctuation marks
Deleted the group and re-created the start button for the correct name on the Strat screen
